src/Console/Commands/BassetInstall.php

```diff
@@ -37,37 +36,40 @@ public function handle(): void
     {
         $this->components->info('Installing Basset');
 
-        // create symlink
-        $this->createSymLink();
-
         if (! $this->option('no-check')) {
             $this->checkBasset();
         }
 
-        // check if artisan storage:link command exists
-        $this->addComposerCommand();
+        // add basset folder to .gitignore
+        $this->addGitIgnore();
+
+        $this->addComposerCommands();
 
         $this->newLine();
         $this->info('  Done');
     }
 
-    /**
-     * Create symlink logic.
-     *
-     * @return void
-     */
-    private function createSymLink(): void
+    private function addGitIgnore()
     {
-        $message = 'Creating symlink';
+        $message = 'Adding public/'.config('backpack.basset.path').' to .gitignore';
 
-        try {
-            $this->callSilent('storage:link');
-            $this->components->twoColumnDetail($message, '<fg=green;options=bold>DONE</>');
-        } catch (Exception $e) {
-            $this->components->twoColumnDetail($message, '<fg=red;options=bold>ERROR</>');
-            $this->line('  <fg=gray>│ '.$e->getMessage().'</>');
-            $this->newLine();
+        if (! file_exists(base_path('.gitignore'))) {
+            $this->components->task($message, function () {
+                file_put_contents(base_path('.gitignore'), 'public/'.config('backpack.basset.path'));
+            });
+
+            return;
         }
```

Comment:
Wait. Why do we add the basset path to `.gitignore` on installation? Wasn't the whole point of publishing to the `public` directory... to have the assets in source control?

Reply:
Hey Cristian, my understanding from last iteration we had is that we would keep the "current behavior" as the default, but removing from `.gitignore` and pushing the public assets to the repo would work as if they were regular files, namely only "changed files" will appear in the diff. 

We can change the default, but for that I would argue that we need a way to "don't load" some assets (things you are not using). I thought about `Basset::map('asset-name', null)` to don't load that specific asset, but it seems cumbersome, so maybe `Basset::dontLoad(['asset-1', 'asset-2'])` would be better (defined in the same `OverrideAssets` class). 

But there is also another issue, if the basset isn't "named", so a @bassetBlock or something similar we would always load it, and I think that's not the desired behavior too. eg:
4 wysiwyg editors. they all load the "editor script" and the "editor initialization code". You only use 1 of them, so you could easily do `Basset::dontLoad(['editor-1', 'editor-2', 'editor-3'])`. That wouldn't load those editors from cdns, but the "initialization scripts" for those editors would still be loaded. Similar issue with the CSS. 

I think one reasonable solution would be to have "grouped assets", so that the `editor.js`, `editor.css` and the `@bassetBlock('editor-initialization')` could be disable in one go by disabling the whole group. This will work if we have "related code blocks/urls" in different files.

If we wanted to simplify things as everything is usually on the same file, we could add `excluded_views` that would not cache anything inside those files.  

I think this is kind of a must when you are publishing assets to your repo, you wouldn't want files that you are not using, either for consistency or for saving space/computation time. 

I didn't implemented nothing like that yet, maybe we can talk a bit about it and decide if it's worth to do right now and how to do it. 👍

Reply:
After further review, I agree with your decision here - `.gitignore` will work best in 99% of the cases, so we keep it as the default, yes.

For the assets-committed-to-git use case... I understand why a new function would be needed, yes. I don't think it's urgent to create one, but here are my thoughts:

```php
Basset::map('asset-name', null); 
// I like it, it simple and clean;

Basset::dontLoad(['editor-1', 'editor-2', 'editor-3']); 
// I don't like the "dont", and don't like the "load" - does that mean it doesn't cache it or what? not explicit enough;

// maybe we can do something like
Basset::ignore('asset-name');
// but this still doesn't answer the whole question... what will happen... 
// - will it not get cached on disk?
// - will it not get loaded onpage at all? or will it be loaded from the CDN?
```

Regarding grouping, I would NOT add grouping. That will add a lot of complexity to our code, for a 1% scenario. It would probably be easier to just add something like this, to allow vendors to create namespaces for names:

```
Basset::ignore('ckeditor.*'); 
```

But even that... to be honest, would add complexity, and I don't know if it's worth it at this point.

--

If you think we should do this... please add an issue for it with our best thoughts... but personally I don't think this is URGENT nor IMPORTANT so I wouldn't spend time on this right now.